src/candid_rpc.rs

```diff
@@ -30,23 +30,24 @@ impl RpcTransport for CanisterTransport {
         METADATA.with(|m| m.borrow().get().nodes_in_subnet)
     }
 
-    fn resolve_api(provider: &RpcNodeProvider) -> Result<RpcApi, ProviderError> {
-        use RpcNodeProvider::*;
+    fn resolve_api(provider: &RpcService) -> Result<RpcApi, ProviderError> {
+        use RpcService::*;
         let (chain_id, hostname) = match provider {
-            Ethereum(provider) => (
+            EthMainnet(provider) => (
                 ETH_MAINNET_CHAIN_ID,
                 match provider {
-                    EthereumProvider::Ankr => "rpc.ankr.com",
-                    EthereumProvider::BlockPi => "ethereum.blockpi.network",
-                    EthereumProvider::Cloudflare => "cloudflare-eth.com",
+                    EthMainnetService::Ankr => "rpc.ankr.com",
+                    EthMainnetService::BlockPi => "ethereum.blockpi.network",
+                    EthMainnetService::PublicNode => "ethereum.publicnode.com",
+                    EthMainnetService::Cloudflare => "cloudflare-eth.com",
```

Comment:
How about defining some constant string slices and collect them in one place (`constants.rs `or maybe a separate file like `domains.rs`...)?